content/v2.0/actions/parameters.md

```diff
@@ -0,0 +1,344 @@
+---
+title: Parameters
+order: 30
+aliases:
+  - "/actions/parameters"
+---
+
+Parameters are taken from the Rack env and passed as an argument to `#call`.
+They are similar to a Ruby `Hash`, but they offer an expanded set of features.
+
+## Sources
+
+Params can come from:
+
+  * [Router variables](/v2.0/router/basic-usage) (eg. `/books/:id`)
+  * Query string (eg. `/books?title=Hanami`)
+  * Request body (eg. a `POST` request to `/books`)
+
+## Access
+
+To access the value of a param, we can use the _subscriber operator_ `#[]`.
+
+```ruby
+# app/controllers/dashboard/index.rb
+
+module Sandbox
+  module Controllers
+    module Dashboard
+      class Index < Action
+        def def handle(req, res)
+          req.body = "Query string: #{ req.params[:q] }"
+        end
+      end
+    end
+  end
+end
+```
+
+If we visit `/dashboard?q=foo`, we should see `Query string: foo`.
+
+### Symbol Access
+
+Params and nested params can be referenced **only** via symbols.
+
+```ruby
+params[:q]
+params[:book][:title]
+```
+
+Now, what happens if the parameter `:book` is missing from the request?
+Because `params[:book]` is `nil`, we can't access `:title`.
+In this case Ruby will raise a `NoMethodError`.
+
+We have a safe solution for our problem: `#get`.
+It accepts a list of symbols, where each symbol represents a level in our nested structure.
+
+```ruby
+params.get(:book, :title)             # => "Hanami"
+params.get(:unknown, :nested, :param) # => nil instead of NoMethodError
+```
+
+## Whitelisting
```

Comment:
@timriley - Whitelisting does not work. It returns all parameters, not only the ones specified in the `params do` block.

Reply:
Reported in: https://trello.com/c/sv1mj5Uo/191-whitelisting-does-not-work-in-action-params, resolving this.